src/main/java/com/fingerprint/Sealed.java

```diff
@@ -0,0 +1,158 @@
+package com.fingerprint;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fingerprint.model.EventResponse;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.zip.Inflater;
+import java.util.zip.InflaterInputStream;
+
+public class Sealed {
+    public enum DecryptionAlgorithm {
+        AES_256_GCM
+    }
+
+    public static class DecryptionKey {
+        private final byte[] key;
+        private final DecryptionAlgorithm algorithm;
+
+        public DecryptionKey(byte[] key, DecryptionAlgorithm algorithm) {
+            this.key = key;
+            this.algorithm = algorithm;
+        }
+    }
+
+    public static class UnsealAggregateException extends Exception {
+        private final List<UnsealException> unsealExceptions = new ArrayList<>();
+
+        public UnsealAggregateException() {
+            super("Failed to unseal with all decryption keys");
+        }
+
+        public void addUnsealException(UnsealException exception) {
+            unsealExceptions.add(exception);
+        }
+
+        public List<UnsealException> getUnsealExceptions() {
+            return unsealExceptions;
+        }
+    }
+
+    public static class InvalidSealedDataException extends IllegalArgumentException {
+        public InvalidSealedDataException() {
+            super("Invalid sealed data");
+        }
+    }
+
+    public static class InvalidSealedDataHeaderException extends IllegalArgumentException {
+        public InvalidSealedDataHeaderException() {
+            super("Invalid sealed data header");
+        }
+    }
+
+    public static class UnsealException extends Exception {
+        public final DecryptionKey decryptionKey;
+
+        public final Exception exception;
+
+        public UnsealException(String message, DecryptionKey decryptionKey, Exception exception) {
+            super(message);
+            this.decryptionKey = decryptionKey;
+            this.exception = exception;
+        }
+    }
+    private static final byte[] SEAL_HEADER = new byte[]{(byte) 0x9E, (byte) 0x85, (byte) 0xDC, (byte) 0xED};
+    private static final int NONCE_LENGTH = 12;
+    private static final int AUTH_TAG_LENGTH = 16;
+
+    public static byte[] unseal(byte[] sealed, DecryptionKey[] keys) throws IllegalArgumentException, UnsealAggregateException {
+        if (!Arrays.equals(Arrays.copyOf(sealed, SEAL_HEADER.length), SEAL_HEADER)) {
+            throw new InvalidSealedDataHeaderException();
+        }
+
+        UnsealAggregateException aggregateException = new UnsealAggregateException();
```

Comment:
I don't think that would work here. We need a fresh instance on every method call in order to retain all errors thrown in current decryption operation. With static instance we would just keep adding new errors to the same instance over and over again. Unless I understand you wrong, sorry but I'm not super familiar with Java and it's practices 🤔

Reply:
Ah, I wasn't aware of that, thanks! I've changed it to use `addSuppressed` method.

> btw,
in case of wrong DecryptionKey you are saving exception an try with a next key from the keys array,
but in case of wrong algorithm in a key you are throwing IllegalArgument and don't try rest of keys.
Does it makes sense to aggregate IllegalArguments as well?

I'm not sure, these seem like two separate cases. 
When invalid keys is passed, we want to attempt decryption with other keys because this specific key could be rotated.
Passing invalid algorithm seems like a valid case to throw IllegalArgumentException and not iterate over remaining keys, because it's incorrect usage of the method. WDYT?